### ISSUE TITLE
Using Push outside of Live

### ISSUE THREAD

**Issue by PierBover** [gluon/AbletonLive9_RemoteScripts#28]
Hi

This is not really an issue, but I don't know where else to ask...

I'm considering using the python files for Push without Live, for the purpose of using Push with other applications like Traktor. I could use Push as a simple midi controller, but it would be awesome if we could send data back to Push and display info in the screens from Traktor, or any other software.

I haven't tried anything yet, but the python scripts import all sorts of libraries like `ableton.v2...`.

Do you know if those python libraries are available?

Do you think if we had access to those Ableton libraries we could for example build a python app or service to bridge between say Traktor and Push directly?

Thanks in advance

**Comment by dataf1ow** [comment]
The python files that work to integrate hardware controllers like the Push with Live use the Live API, which is NOT available for other programs to use. The libraries that scripts import are part of the Live distribution, and are not openly available. You'll have to build it from scratch unfortunately, if you want to have some sort of scriptable MIDI interpreter/translator. 

If you're using traktor, you can send info back, but I doubt that you'll be able to address the screen, I don't think traktor allows for sysex messages being sent out.

**Comment by PierBover** [comment]
Thanks for your answer. It's a shame Ableton isn't releasing any SDK for using Push outside of Live.

I'm not sure about sysex, but Traktor does send a ton of data out. Anyway I was only using Traktor as an example. The applications for using Push with custom software are endless in say museums, science labs, robotics, etc. It's an amazing input device with many possibilities outside of music production.

**Comment by dataf1ow** [comment]
Yeah, an SDK would be nice, but as we've heard for years, they just don't have the resources/desire to document and support it. It would be a lot of work. 

At least they don't prevent users from tinkering with the API though.

**Comment by gluon** [comment]
Push got a user mode, afaik.

Julien

**Comment by gluon** [comment]
Are you kidding?
They are so numerous.
And they maintain that for companies making remote surfaces.

Julien

**Comment by alandrees** [comment]
you could probably very easily reverse engineer the push-side of the scripting into something that's more portable

**Comment by dataf1ow** [comment]
Whos kidding who? I didn't mean there aren't resources, but they aren't
'official'm and they aren't released, documented, or supported publicly by
Ableton...

On Tue, Jan 12, 2016 at 1:22 PM, Alan Drees <notifications@github.com>
wrote:

> you could probably very easily reverse engineer the push-side of the
> scripting into something that's more portable
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gluon/AbletonLive9_RemoteScripts/issues/28#issuecomment-171001530>
> .
>



-- 
*Evan Bogunia* - Sound_designer.composer.educator
ebogunia@gmail.com
574-261-5146

**Comment by alandrees** [comment]
As far as I remember when I was talking to an Ableton rep about it (this was years ago, maybe it's changed by now), but all the remote scripts are actually written by Ableton controller script devs.  The controller companies give them specs and the script developers actually write and maintain all of them.  The conversation I had was specifically in relation to the Novation Launchpad.

**Comment by gluon** [comment]
"they just don't have the resources", I answering to that.
anyway, we cannot change them, but we can do so much things with what they left in these folders, already.

**Comment by dataf1ow** [comment]
I agree! I shouldn't have said 'they don't have the resources' - It
should've been 'they don't want to use the resources'.


On Tue, Jan 12, 2016 at 1:29 PM, Julien Bayle <notifications@github.com>
wrote:

> "they just don't have the resources", I answering to that.
> anyway, we cannot change them, but we can do so much things with what they
> left in these folders, already.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gluon/AbletonLive9_RemoteScripts/issues/28#issuecomment-171003521>
> .
>



-- 
*Evan Bogunia* - Sound_designer.composer.educator
ebogunia@gmail.com
574-261-5146

**Comment by CIPop** [comment]
For future reference: Ableton released the Push2 docs here: https://github.com/Ableton/push-interface
